modules/local/nf-core-modified/iqtree.nf

```diff
@@ -39,29 +39,31 @@ process IQTREE {
         -nt AUTO \\
         -ntmax ${task.cpus} \\
         -mem \$memory \\
-        -m LG+F+G4 \\
-        $args \\
-        $fconst_args
-
-    # Identify the best number of threads
-    nt=\$(grep "BEST NUMBER" *.log | sed "s/.*: //g")
-
-    # Rename it and clean up
-    mv *.treefile guidetree.treefile
-    rm *fa.*
-
-    iqtree \\
-        -s $alignment \\
-        -nt \$nt \\
-        -mem \$memory \\
-        -m LG+C40+F+G4 \\
-        -ft guidetree.treefile \\
-        $args \\
-        $fconst_args
-
-    # Clean up
-    rm ./guidetree.treefile
+        -m $model \\
+        $args
 
+    # check if we're running a PMSF approximation - if so, do the following,
+    # treating the tree inferred above as a guide tree
+    if [ "$pmsf_model" != "none" ]; then
+        # Identify the best number of threads
+        nt=\$(grep "BEST NUMBER" *.log | sed "s/.*: //g")
```

Comment:
hate this lol

Reply:
Me too - we should be able to use the normal "auto" method for detecting number of threads, but it fails for whatever reason when running the PMSF approximation. That bug is internal to IQ-Tree. So, this is just a workaround to still use a sensible number of threads from the previous step. 

I might end up looking into whether the tree estimation can actually slow down from overspecification of threads, or if it's just a question of diminishing returns. If the latter, I could imagine just specifying a fixed number of threads (as a parameter, with the auto method as default) if we're running on tower for instance.